packages/core/src/App/Containers/Redirect/redirect.jsx

```diff
@@ -2,6 +2,7 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import { withRouter } from 'react-router-dom';
 import { loginUrl, routes, PlatformContext } from '@deriv/shared';
+import { getAllowedLanguages, currentLanguage, getURL } from 'Utils/Language';
```

Comment:
Question: in general how does it happen that the currentLanguage is not supported on this stage? Does it come from deriv.com?

Reply:
Not really. In deriv.com we have a constant for supported language: https://github.com/binary-com/deriv-com/blob/cc562c0d98bd83826d4d656badc181d5af754ce5/src/common/constants.js#L38 which will fall back to english if it is not within the supported language: https://github.com/binary-com/deriv-com/blob/cc562c0d98bd83826d4d656badc181d5af754ce5/src/common/utility.js#L85

Reply:
Hmm, seems like the email users are getting contains the unsupported language in URL and also is messed up with some wordings in english, some in ![unsupported language.](https://user-images.githubusercontent.com/71861519/123024959-80c4a880-d40c-11eb-8155-d72cc3b9d000.png). I think the email template should be changed.

Reply:
seems like this is the root cause 😆

Reply:
I discussed today a bit with Akmal on this for a solution. For now, I will see to apply a global fix, so that if the user tries to access different languages through the link (for example, also typing the 'lang=RU' in the link) it will default back to english. Aside from that I will have a talk with Mamali/Amal to see how the template can be fixed.

Thanks a lot for the feedback guys! :)